v2/pkg/subscraping/sources/facebook/ctlogs.go

```diff
@@ -0,0 +1,196 @@
+package facebook
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/projectdiscovery/gologger"
+	"github.com/projectdiscovery/retryablehttp-go"
+	"github.com/projectdiscovery/subfinder/v2/pkg/subscraping"
+	errorutil "github.com/projectdiscovery/utils/errors"
+	"github.com/projectdiscovery/utils/generic"
+	urlutil "github.com/projectdiscovery/utils/url"
+)
+
+// source: https://developers.facebook.com/tools/ct
+// api-docs: https://developers.facebook.com/docs/certificate-transparency-api
+// ratelimit: ~20,000 req/hour per appID https://developers.facebook.com/docs/graph-api/overview/rate-limiting/
+
+var (
+	domainsPerPage = "1000"
+	authUrl        = "https://graph.facebook.com/oauth/access_token?client_id=%s&client_secret=%s&grant_type=client_credentials"
```

Comment:
Unrelated and optional: somehow I think that string holding fmt formats should be more explicit or easily identifiable, with naming or maybe defining a new alias type? What do you think of implementing an enriched StringFmt type like:

```
// with naming
var variableNameFmt = "xxx%s"

// or with alias type
// var xxx StringFmt = "xxx%s"
// xxx.Printf(arg[0], arg[1])

type StringFmt string

func (s StringFmt) Printf(args ...interface{}) { ... }
```